napari/_qt/dialogs/_tests/test_installer_process.py

```diff
@@ -0,0 +1,219 @@
+import os
+import sys
+import time
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pytest
+from qtpy.QtCore import QProcessEnvironment
+
+from napari._qt.dialogs.qt_package_installer import (
+    AbstractInstallerTool,
+    CondaInstallerTool,
+    InstallerQueue,
+    InstallerTools,
+    PipInstallerTool,
+)
+
+if TYPE_CHECKING:
+    from virtualenv.run import Session
+
+
+@pytest.fixture
+def tmp_virtualenv(tmp_path) -> 'Session':
+    virtualenv = pytest.importorskip('virtualenv')
+
+    cmd = [str(tmp_path), '--no-setuptools', '--no-wheel', '--activators', '']
+    return virtualenv.cli_run(cmd)
+
+
+def conda_exe():
```

Comment:
There is a double definition of this discovery process. I prefer to call `CondaInstallerTool.executable()` implementation, that currently is not tested.

Reply:
This one has some extra logic to handle CI environments (`$CONDA` is defined by GHA, but it's not an "official" env var). That said, no good reason. I will consolidate!